src/concurrency/thread.rs

```diff
@@ -192,6 +202,8 @@ impl Time {
     fn get_wait_time(&self) -> Duration {
```

Comment:
You'll probably have to pass the `Machine` to this function so that it can access the virtual clock to compute the wait time.

Reply:
Yeah I thought it would be better to keep `get_wait_time` signature unaltered and keep a reference to the global clock in the `Time`. I assumed borrowing issues would appear by having to pass a reference the machine or part of it as an argument but I can try and see what happens.

Reply:
I'd prefer explicitly borrowing here over implicitly sharing an `Arc` in a bunch of hard-to-track places.